www/include/configuration/configObject/contactgroup/DB-Func.php

```diff
@@ -190,26 +194,35 @@ function insertContactGroup($ret)
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
+    $cgActivate = filter_var($ret["cg_activate"]["cg_activate"], FILTER_SANITIZE_STRING); //enum
 
-    $rq = "INSERT INTO `contactgroup` (`cg_name`, `cg_alias`, `cg_comment`, `cg_activate`) ";
-    $rq .= "VALUES ('" . CentreonDB::escape($ret["cg_name"]) . "', '" . CentreonDB::escape($ret["cg_alias"]) .
-        "', '" . CentreonDB::escape($ret["cg_comment"]) . "', '" . $ret["cg_activate"]["cg_activate"] . "')";
-    $pearDB->query($rq);
+    $stmt = $pearDB->prepare(
+        "INSERT INTO `contactgroup` (`cg_name`, `cg_alias`, `cg_comment`, `cg_activate`)
+        VALUES (:cgName, :cgAlias, :cgComment, :cgActivate)"
+    );
+
+    $stmt->bindValue(':cgName', $cgName, \PDO::PARAM_STR);
+    $stmt->bindValue(':cgAlias', $cgAlias, \PDO::PARAM_STR);
+    $stmt->bindValue(':cgComment', $cgComment, \PDO::PARAM_STR);
```

Comment:
what will happen if comment contains only illegal chars ?
filter_var will return false and bindValue will try to bind false
same for other fields

Reply:
IMHO : you'll need to check that each of the sanitized variable are different from false